jax/interpreters/pxla.py

```diff
@@ -1832,6 +1832,49 @@ def _pmap_lowering(ctx, *in_nodes, axis_name,
 # ------------------- xmap -------------------
 
 class Mesh(ContextDecorator):
+  """Declare the hardware resources available in the scope of this manager.
+
+  In particular, all ``axis_names`` become valid resource names inside the
+  managed block and can be used e.g. in the ``axis_resources`` argument of
+  :py:func:`xmap`.
+
+  If you are compiling in multiple threads, make sure that the
```

Comment:
I wonder if this is worth including. Is there ever a context manager that works any other way?

Reply:
It lead to some confusion before, so I think its fine to include this.